src/relayFeeCalculator/relayFeeCalculator.ts

```diff
@@ -64,7 +119,39 @@ export class RelayFeeCalculator {
   // Note: these variables are unused now, but may be needed in future versions of this function that are more complex.
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   async capitalFeePercent(_amountToRelay: BigNumberish, _tokenSymbol: string): Promise<BigNumber> {
-    return toBNWei(this.capitalCostsPercent / 100);
+    // V0: Charge fixed capital fee
+    const defaultFee = toBNWei(this.capitalCostsPercent / 100);
+
+    // V1: Charge fee that scales with size. This will charge a fee % based on a linear fee curve with a "kink" at a
+    // cutoff in the same units as _amountToRelay. Before the kink, the fee % will increase linearly from a lower
+    // bound to an upper bound. After the kink, the fee % increase will be fixed, and slowly approach the upper bound
+    // for very large amount inputs.
+    if (this.capitalCostsConfig[_tokenSymbol]) {
+      const config = this.capitalCostsConfig[_tokenSymbol];
+      // Scale amount "y" to 18 decimals
+      const y = toBN(_amountToRelay).mul(toBNWei("1", 18 - config.decimals));
+      // At a minimum, the fee will be equal to lower bound fee * y
+      const minCharge = config.lowerBound.mul(y).div(fixedPointAdjustment);
+
+      // Charge an increasing marginal fee % up to min(cutoff, y). If y is very close to the cutoff, the fee %
+      // will be equal to half the sum of (upper bound + lower bound).
+      const yTriangle = min(config.cutoff, y);
+
+      // triangleSlope is slope of fee curve from lower bound to upper bound.
+      // triangleCharge is interval of curve from 0 to y for curve = triangleSlope * y
+      const triangleSlope = config.upperBound.sub(config.lowerBound).mul(fixedPointAdjustment).div(config.cutoff);
+      const triangleHeight = triangleSlope.mul(yTriangle).div(fixedPointAdjustment);
+      const triangleCharge = triangleHeight.mul(yTriangle).div(toBNWei(2));
+
+      // For any amounts above the cutoff, the marginal fee % will not increase but will be fixed at the upper bound
+      // value.
+      const yRemainder = max(toBN(0), y.sub(config.cutoff));
+      const remainderCharge = yRemainder.mul(config.upperBound.sub(config.lowerBound)).div(fixedPointAdjustment);
+
+      return minCharge.add(triangleCharge).add(remainderCharge).mul(fixedPointAdjustment).div(y);
```

Comment:
@cc7768 can you validate this math?

Reply:
I didn't carefully check things that were big number specific (i.e. decimal places and fixed point adjustments), but the  mathematical operations are all exactly what I would expect.

Let me know if you want me to give the other things a closer look